### DISCUSSION TITLE
Serialization strategy

### DISCUSSION THREAD

**Discussion by oleganza** [tact-lang/spec#7] [Ideas]
I suggest moving the discussion of serialization/encoding approaches to this issue. The way we approach serialization of messages affects:
* the interop with TL/B and FunC, 
* partial decoding of TL/B messages from Cells (important for performance),
* ABI stability story,
* design of unions ([see discussion](https://github.com/tact-lang/spec/discussions/4#discussioncomment-2707743)),  
* design of annotations.

Let's put together all the thoughts, pros/cons here.

**Comment by cmxog** [comment]
Capturing a thought from a conversation with @p0lunin:

We might solve some [problems with serialization](https://github.com/tact-lang/spec/discussions/4#discussioncomment-2708756) by designing the serialization API in such a way that it is not only parameterized by its input (what we're serializing) but also by an output parameter as well (what we're deserializing with?)
